lib/WorkflowEager.groovy

```diff
@@ -85,4 +85,38 @@ class WorkflowEager {
 
     }
 
+    def public static ArrayList addNewMetaFromAttributes( ArrayList row, Object source_attributes, Object target_attributes, boolean remove = false) {
+        def meta = row[0]
+        def meta2 = [:]
+
+        // Read in target and source attributes and create a mapping between them
```

Comment:
Coud you add as docs an example of exactly what it does,

I presume it would be:

```
[ [meta.foo, meta.bar], file ]
// to
[meta.foo], [meta.bar], file
```

for example, I personally don't find the function name descriptive in  way that makes me interpret that so it ws confusing me initially when seeing it used in teh pipeline

Reply:
Added an explanation comment above the function showing how the channel changes depending on the way the function was called.